### PR TITLE
chore: Default dev mode to production cloud rather than staging

### LINKED ISSUE #23020: Point yarn dev to production by default
In order to allow developers to work with accurate data in yarn dev, we want to point Cypress to the production environment of the Dashboard by default.

Right now when you hit login in the app (or launchpad) it defaults to https://dashboard-staging.cypress.io/organizations/new which is **staging**. Let's point everything at **production**, which is https://dashboard.cypress.io. by default so we can use the dashboard and ACI features more actively. 

One other thing we do is you can launch Cypress with `CYPRESS_INTERNAL_ENV` or `CYPRESS_INTERNAL_CLOUD_ENV` with either `staging` or `production`. This might help you find places in the code we have diverging behavior.

AC:

- when logging in, default to production
- should show latest production runs when logging in
  - You can see those [here](https://dashboard.cypress.io/projects/ypt4pf/runs?branches=%5B%7B%22label%22%3A%22develop%22%2C%22suggested%22%3Afalse%2C%22value%22%3A%22develop%22%7D%5D&committers=%5B%5D&flaky=%5B%5D&page=1&status=%5B%5D&tags=%5B%5D&timeRange=%7B%22startDate%22%3A%221970-01-01%22%2C%22endDate%22%3A%222038-01-19%22%7D)
  - this is controlled by `projectId` in `cypress.config.js`

### PR DESCRIPTION
**Pull request by astone123**
- Closes #23020

### User facing changelog

n/a

### Additional details

This PR defaults to using the production instance of cloud when the app is running in development mode. Before, it was using the staging instance. You can still target staging by setting `CYPRESS_INTERNAL_ENV=staging`.

This will allow us to see real CI runs for specs that are in our cypress packages while we develop

### Steps to test

1. Run the app with this branch
2. Go to the Runs tab and click on a run
3. Verify that you are taken to https://dashboard.cypress.io/ rather than https://dashboard-staging.cypress.io/
4. Verify that you can see runs in the latest runs in the specs list under the latest runs column
5. Set `CYPRESS_INTERNAL_ENV=staging` and re-run the application
6. Repeat step 2 and verify that you are now taken to https://dashboard-staging.cypress.io/

### PR Tasks

- [na] Have tests been added/updated?
- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/app/cypress.config.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ const CYPRESS_INTERNAL_CLOUD_ENV = getenv('CYPRESS_INTERNAL_CLOUD_ENV', process.
 const CYPRESS_INTERNAL_DEV_PROJECT_ID = getenv('CYPRESS_INTERNAL_DEV_PROJECT_ID', process.env.CYPRESS_INTERNAL_DEV_PROJECT_ID || 'sehy69')
 
 export default defineConfig({
-  projectId: CYPRESS_INTERNAL_CLOUD_ENV === 'staging' ? 'ypt4pf' : CYPRESS_INTERNAL_DEV_PROJECT_ID,
+  projectId: CYPRESS_INTERNAL_CLOUD_ENV === 'staging' || CYPRESS_INTERNAL_CLOUD_ENV === 'production' ? 'ypt4pf' : CYPRESS_INTERNAL_DEV_PROJECT_ID,
```
Comment (astone123) [inline packages/app/cypress.config.ts:9]: Our project ID is the same for staging and production, so we should use the same ID for both envs

**Comment by rockindahizzy** [comment]
I don't think it's an issue, but wanted to bring it up just in case. Seems more like bad data.

When I click the first run in the list I get:
```bash
[cy:open:dev:7966]: Error: Invalid URL
[cy:open:dev:7966]:     at openExternal (/Users/rocky/workspace/cypress/packages/server/lib/gui/links.ts:8:29)
```

Running with `CYPRESS_INTERNAL_ENV=staging yarn cypress:open`

**Comment by rockindahizzy** [comment]
Actually, this might be legit. I got it when pointing to prod as well. I am looking at e2e in the launchpad project

```bash
[cy:open:dev:9790]: Error: Invalid URL
[cy:open:dev:9790]:     at openExternal (/Users/rocky/workspace/cypress/packages/server/lib/gui/links.ts:8:29)

```

**Comment by astone123** [comment]
@rockindahizzy yeah I'm seeing those on `develop` as well, so it seems like a different issue to me

**Review by rockindahizzy — APPROVED** [review]
Works as expected!

**Comment by lmiller1990** [comment]
Runs page is fine, why don't these guys show up?

image

**Review by lmiller1990 — COMMENTED** [review]
Quick q, I want to see the runs status on the main page if possible :thinking:

**Comment by mike-plummer** [comment]
> Runs page is fine, why don't these guys show up?

@lmiller1990 The "Latest Runs" data is branch-specific, and there haven't been any runs recorded on his branch thus the gray placeholders. If this was a fresh branch with no commits of its own it would fall back to showing runs from the source branch, but this branch has multiple commits.

**Comment by astone123** [comment]
> > Runs page is fine, why don't these guys show up?
> 
> @lmiller1990 The "Latest Runs" data is branch-specific, and there haven't been any runs recorded on his branch thus the gray placeholders. If this was a fresh branch with no commits of its own it would fall back to showing runs from the source branch, but this branch has multiple commits.

Shouldn't there be some runs recorded on this branch though? CI has run on a few commits for this branch so we should expect some runs have been recorded in prod, right?

**Comment by astone123** [comment]
Moving this back into draft

**Review by emilyrohrbough — APPROVED** [review]

**Review by lmiller1990 — APPROVED** [review]
@astone123 runs page looks good, do we know why I can't see any info on the main specs page for `app`? I merged this into develop but still couldn't see any -- this seems incorrect, I wonder why I can't see anything there? Do we need to connect up a project via the dashboard or something? :thinking:

**Pull request merged**